io/net/virt-net/Network_virtualization_Backing_device_add_remove.py

```diff
@@ -0,0 +1,244 @@
+              ' -F lpar_id'
+        self.lpar_id = self.run_command(cmd)[-1]
+        cmd = 'lssyscfg -m ' + self.server + \
+              ' -r lpar --filter lpar_names=' + self.vios_name + \
+              ' -F lpar_id'
+        self.vios_id = self.run_command(cmd)[-1]
+        self.backing_devices = "sriov/%s/%s/%s/%s/%s"\
+                               % (self.vios_name, self.vios_id,
+                                  self.sriov_adapter_id, self.sriov_port,
+                                  self.bandwidth)
+
+    def login(self, ip, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        p = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        p.force_password = True
+
+        p.login(ip, username, password)
+        p.sendline()
+        p.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        p.set_unique_prompt()
+        p.prompt(timeout=60)
+        self.pxssh = p
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        c = self.pxssh
+        c.sendline(command)
+        c.expect("\n")  # from us
+        c.expect(c.PROMPT, timeout=timeout)
+        output = c.before.splitlines()
+        c.sendline("echo $?")
+        c.prompt(timeout)
+        return output
+
+    def test(self):
+        '''
+        Start rsct services and performs add and remove operations
+        of Network virtualized ports
+        '''
+        try:
+            for svc in ["rsct", "rsct_rm"]:
+                process.run('startsrc -g %s' % svc, shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Starting service %s failed", svc)
+
+        output = process.system_output("lssrc -a", ignore_status=True,
+                                       shell=True, sudo=True)
+        if "inoperative" in output:
+            self.fail("Failed to start the rsct and rsct_rm services")
+
+        for _ in range(self.count):
+            self.network_virtualization_backing_device_add()
+            self.network_virtualization_backing_device_remove()
+
+    def network_virtualization_backing_device_remove(self):
+        '''
+        Network virtualized port remove operation
+        '''
+        self.changehwres(self.server, 'r', self.lpar_id, self.slot_num,
+                         self.backing_devices, 'remove')
+        output = self.listhwres(self.server, self.lpar)
+        # To Do: parse the output for pass/fail
```

Comment:
The output needs to be taken before add/remove and after add/remove.
Then, the count of sriov needs to be compared.

Reply:
@hathyaga Take the output of listhwres() and find the count. Now do this before and after add/remove.

Reply:
@hathyaga If the output is given, then I can say it it looks fine or not. Not with code snippet alone.

Reply:
The job.log has the o/p.  it lists the backing devices before and after backing device add/remove.

Reply:
My concern is what i do to validate that data in function listhwres(self, server, lpar):
Can you check that?

Reply:
```
Running the command on hmc chhwres -r virtualio --rsubtype vnic -o s -m ltc-zz3 -s 4 --id 15                    -a backing_devices+=sriov/ltc-zz3-vios1/100/1/0/4
Running the command on hmc lshwres -r virtualio -m  ltc-zz3 --rsubtype vnic --level lpar --filter                "lpar_names=ltc-zz3-lp15-vnic-automation" -F slot_num,backing_device_states
[stdout] ['3,"sriov/27004037/1/Operational,sriov/27004039/0/Operational"', '4,"sriov/27004038/1/Operational,sriov/27004043/0/Operational"']
[stdout] 

Running the command on hmc chhwres -r virtualio --rsubtype vnic -o s -m ltc-zz3 -s 4 --id 15                    -a backing_devices-=sriov/ltc-zz3-vios1/100/1/0/4
Running the command on hmc lshwres -r virtualio -m  ltc-zz3 --rsubtype vnic --level lpar --filter                "lpar_names=ltc-zz3-lp15-vnic-automation" -F slot_num,backing_device_states
[stdout] ['3,"sriov/27004037/1/Operational,sriov/27004039/0/Operational"', '4,sriov/27004038/1/Operational']
[stdout]
```

Reply:
The command looks good, I thought we already discussed and finalised on that.
Count on that needs to be taken, as mentioned.

Reply:
I assume here the `output = ['27004038']
Isn't it ?